.github/workflows/ci.yml

```diff
@@ -434,15 +381,7 @@ jobs:
     outputs:
       matrix: ${{ steps.set-matrix.outputs.matrix }}
     steps:
-      - uses: actions/checkout@v3
-        with:
-          fetch-depth: 0 # checkout all commits to be able to determine merge base for GIB
-      - name: Fetch base ref to find merge-base for GIB
-        run: .github/bin/git-fetch-base-ref.sh
-      - uses: actions/setup-java@v3
-        with:
-          distribution: 'zulu'
-          java-version: 17
+      - uses: trinodb/github-actions/checkout-and-setup@6a40cfbfa0ad484d69f5482b29c41fbaac72944e
```

Comment:
This [action](https://github.com/trinodb/github-actions/blob/main/checkout-and-setup/action.yml) has `input.cache`default set to `maven` which is not the same as the [default](https://github.com/actions/setup-java/blob/de1bb2b0c5634f0fc4438d7aa9944e68f9bf86cc/action.yml#L55) from [setup-java](https://github.com/actions/setup-java/tree/v3) and means it's not a simple drop-in replacement as it changes behavior in some jobs. This job is a good example since it assumes there would be no attempt to update the cache with whatever `$RETRY $MAVEN validate` downloads, and makes no attempt to delete the local Maven repo.

Maybe the composite action should remove the whole Maven repo at the end, and only leave it behind if a special input is set? This input would only be set in `maven-checks` job. That way most jobs would not need any action inputs.

Reply:
Thanks, I need to adjust inputs here, this commit is older than the recent work we did around the cache. The action sets the cache to `maven` by default since this is the most common input for most jobs, but we do need to disable it here.

Unfortunately, a composite action cannot register post-job hooks.

Reply:
I explicitly disabled cache in `web-ui-checks` and here in `build-test-matrix` jobs.